tsdb/shard.go

```diff
@@ -472,6 +506,11 @@ func (s *Shard) SeriesKeys(opt influxql.IteratorOptions) (influxql.SeriesList, e
 		return []influxql.Series{{Aux: auxFields}}, nil
 	}
 
+	s.mu.RLock()
```

Comment:
I think this is a good idea. In this case, we can also avoid using a defer. @joelegasse pointed out to me once that a defer has a performance penalty.

Here's the first article I could find about it so I'm not sure how accurate the article is: http://lk4d4.darth.io/posts/defer/